File: flutter/fctconnect/lib/models/FeedData.dart
class FeedData {
  final String id;
  final String text;
  final String user;
  final String url;
  final String profilePic;
  final String timestamp;
  final List<dynamic> likes;

  FeedData({
    required this.id,
    required this.text,
    required this.user,
    required this.url,
    required this.profilePic,
    required this.timestamp,
    required this.likes,
  });

  factory FeedData.fromJson(Map<String, dynamic> json) {
    return FeedData(
      id: json['id'],
      text: json['text'],
      user: json['user'],
      url: json['url'],
      profilePic: json['profilePic'],
      timestamp: json['timestamp'],
      likes:json['likes'],
    );
  }

  get postID => null;
}

File: java&html/src/main/java/pt/unl/fct/di/apdc/fctconnect/util/Feed/FeedData.java
package pt.unl.fct.di.apdc.fctconnect.util.Feed;

import java.util.List;

public class FeedData {

    public String id;
    public String text;
    public String user;
    public String url;
    public String timestamp;
    public List<String> likes;
    public String profilePic;
    public FeedData(){};

    public FeedData(String id, String text, String user, String url, String timestamp, List<String> likes, String profilePic){
        this.id = id;
        this.text = text;
        this.user = user;
        this.url = url;
        this.timestamp = timestamp;
        this.likes = likes;
        this.profilePic = profilePic;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getText() {
        return text;
    }

    public String getUser(){
        return user;
    }

    public String getUrl() {
        return url;
    }

    public String getTimestamp() {
        return timestamp;
    }

    public List<String> getLikes() {
        return likes;
    }

    public String getProfilePic() {
        return profilePic;
    }

    public String setText(){
        return text;
    }

    public String setUser(){
        return user;
    }

    public String getBlobName() {
        return url;
    }

    public void setUrl(String blobName) {
        this.url = blobName;
    }
    public void setTimestamp(String timestamp) {
        this.timestamp = timestamp;
    }
}
